### PR TITLE
feat(ereal): standardize + property coverage for math/functions pow/sqrt/hypot (#950 cat 4)

### LINKED ISSUE #950: Refactor elastic/ereal/math/functions/ (16 files) to standard test pattern
**Parent epic: #944**

## Goal

Refactor the 16 files under `elastic/ereal/math/functions/` to the standard regression-test structure. One function (or tightly related family) per file.

## Files

- classify.cpp -- isnan, isinf, isnormal, fpclassify
- ereal_math_stub_test.cpp -- decide if obsolete; possibly remove
- error_and_gamma.cpp -- erf, erfc, lgamma, tgamma
- exponent.cpp -- exp, exp2, expm1, exp10
- fractional.cpp -- modf, fmod, remainder
- hyperbolic.cpp -- sinh, cosh, tanh, asinh, acosh, atanh
- hypot.cpp -- hypot
- logarithm.cpp -- log, log2, log10, log1p
- minmax.cpp -- min, max
- next.cpp -- nextafter, nexttoward
- numerics.cpp -- abs, copysign, signbit
- pow.cpp -- pow, sqrt-by-pow special cases
- sqrt.cpp -- sqrt, cbrt
- trigonometry.cpp -- sin, cos, tan, asin, acos, atan, atan2
- truncate.cpp -- ceil, floor, trunc, round, rint

## Pattern

See `arithmetic/addition.cpp` post-PR #943. Each file gets the `MANUAL_TESTING` / `REGRESSION_LEVEL_OVERRIDE` scaffold and uses `ReportTestSuiteHeader / ReportTestResult / ReportTestSuiteResults`.

## Coverage per function

Verify against high-precision reference values (MPFR oracle when #944's Phase E lands) for representative inputs. Pre-fuzzer, hand-curated test cases:
- Identity / fixed points (e.g. `exp(0) == 1`, `log(1) == 0`)
- Inverse relations (e.g. `exp(log(x)) == x`)
- Sign symmetries (e.g. `sin(-x) == -sin(x)`)
- Special values (`+0`, `-0`, `inf`, `nan` where applicable)
- Subnormal-input edge cases

## Acceptance

- All 16 files use the standard framework.
- Each test is single-function (function under test only; setup may use constructors / unrelated math).
- Build clean gcc + clang, debug + release.
- `ereal_math_stub_test.cpp` decision documented (keep, rename, or remove).

## Suggested decomposition

This issue is large -- consider splitting into one PR per file or one PR per category (trig, exp/log, hyperbolic, etc.). Reviewer load is the main constraint.

### PR DESCRIPTION
**Pull request by Ravenwater**
## Summary
Category 4 of 5 for the `elastic/ereal/math/functions/` refactor (#950): `pow.cpp`, `sqrt.cpp`, `hypot.cpp`.

Relates to #944 (parent epic, Phase B). Relates to #950 (partial -- category 4 of 5).

## Standardization
- anonymous namespace + single `using namespace sw::universal;`
- `reportTestCases`-gated `std::cout << "    FAIL ..."` (was `std::cerr`)
- `reportTestCases` defaults to `true`
- stripped non-ASCII (`~=`, `^2`, `^3`, `pi`) from comments

## Layer-1 property coverage (oracle-free; self-oracle scoped to #954)
- `pow.cpp` `VerifyPowFuzz`: `pow(x,2)==x*x`, `pow(x,0.5)==sqrt(x)`, exponent law `pow(x,a)*pow(x,b)==pow(x,a+b)`, `pow(x,0)==1`, `pow(x,1)==x`.
- `sqrt.cpp` `VerifySqrtFuzz`: `sqrt(x)^2==x`, `sqrt(x*x)==x`, `cbrt(x)^3==x`, sqrt monotonicity.
- `hypot.cpp` `VerifyHypotFuzz`: `hypot(x,y)^2==x^2+y^2`, `hypot(x,0)==|x|`, symmetry, `hypot >= max(|x|,|y|)` lower bound.

Probing showed these hold exactly or to ~1e-16 (1e-13 tolerance used). **Note:** hypot's value identities are compared on the projected `double`, not structurally -- its internal expansion need not be bit-identical to the single-limb `|x|` even when the values match (an exact `operator==` failed here; tolerance is correct).

## Test Results
| Target | gcc build | gcc test | clang build | clang test |
|--------|-----------|----------|-------------|------------|
| er_math_pow | OK | PASS | OK | PASS |
| er_math_sqrt | OK | PASS | OK | PASS |
| er_math_hypot | OK | PASS | OK | PASS |

## Remaining
5. fractional/error_and_gamma + stub refactor (final PR -> Resolves #950).

## Test plan
- [ ] Fast CI passes (gcc + clang CI_LITE)
- [ ] Promote to ready when satisfied: `gh pr ready <NNN>`

Relates to #950

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Expanded test coverage with property-based fuzzing for hypot, pow, and sqrt to exercise many randomized cases.
  * Added robust numeric comparisons and checks (identity, exponent laws, symmetry, monotonicity, and bounds) to catch subtle inaccuracies.
  * Enabled more verbose per-test reporting and improved diagnostic messages to make failures easier to investigate.

[Review Change Stack](https://app.coderabbit.ai/change-stack/stillwater-sc/universal/pull/977?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: elastic/ereal/math/functions/pow.cpp [hunk 1]
```diff
@@ -299,15 +302,57 @@ namespace sw {
...
+				if (!close_rel(pow(x, half), sqrt(x), 1.0e-13)) {
+					if (reportTestCases) std::cout << "    FAIL pow(x,0.5)==sqrt(x) at x=" << dx << '\n';
+					++nrOfFailedTestCases;
+				}
+				if (!close_rel(pow(x, Real(1.5)) * pow(x, half), pow(x, two), 1.0e-13)) {
+					if (reportTestCases) std::cout << "    FAIL pow exponent law at x=" << dx << '\n';
+					++nrOfFailedTestCases;
+				}
+				if (pow(x, Real(0.0)) != one || pow(x, one) != x) {
+					if (reportTestCases) std::cout << "    FAIL pow fixed points at x=" << dx << '\n';
```
[the remark this answers is not in the document]
Comment (Ravenwater) [reply elastic/ereal/math/functions/pow.cpp]: Fixed in 52adf617 -- pow(x,0)==1 and pow(x,1)==x now use close_rel(1e-13) instead of exact ==, consistent with the other pow checks. Verified on gcc-13 and clang-18.
